src/services/services.ts

```diff
@@ -1198,12 +1198,13 @@ namespace ts {
             return sourceFile;
         }
 
-        function synchronizeHostData(): void {
+        function synchronizeHostData(bindNodes = true): void {
             // perform fast check if host supports it
             if (host.getProjectVersion) {
                 const hostProjectVersion = host.getProjectVersion();
                 if (hostProjectVersion) {
                     if (lastProjectVersion === hostProjectVersion && !host.hasChangedAutomaticTypeDirectiveNames) {
+                        bindNodesIfNecessary();

```

Comment:
This doesnt seem correct especially from tsserver perspective given that it may not even do this check because it also keepts track of "isDirty" bit

Reply:
I'm happy to scrap this whole change - my actual goal is to not do any binding in a `--syntaxOnly` server (where we ask only syntax-based questions).  If you have an idea for a better way to do that, I'm very interested.

Reply:
It would be nice if the same applied to projects with disabled language services, but that's much lower priority.

Reply:
It seems to me then that Language Service creation itself knows whether to do getTypeChecker or not, and createLanguageService itself can take that parameter. 
For projects that enable/disable language service can flip that switch and project's need to be marked dirty when their language service gets either enabled or disabled to take that into affect. That way this check is in only one place (program.getTypeChecker()) 

Also https://github.com/Microsoft/TypeScript/blob/master/src/services/documentRegistry.ts#L177 if that is not changed, you would need to handle that case, because thats the first time sourceFile is created? So we wont have parentPos set for anything that gets created in the first pass ? So may be you would need to pass your switch to aquireAndUpdate to ensure that first sourceFile also has parent positions set.

Reply:
> It seems to me then that Language Service creation itself knows whether to do getTypeChecker or not, and createLanguageService itself can take that parameter.
For projects that enable/disable language service can flip that switch and project's need to be marked dirty when their language service gets either enabled or disabled to take that into affect. That way this check is in only one place (program.getTypeChecker())

it is a bit strange that we create a languageService object, but tell it to be disabled. I think this is a feature of the server.

> Also https://github.com/Microsoft/TypeScript/blob/master/src/services/documentRegistry.ts#L177 if that is not changed, you would need to handle that case, because thats the first time sourceFile is created? So we wont have parentPos set for anything that gets created in the first pass ? So may be you would need to pass your switch to aquireAndUpdate to ensure that first sourceFile also has parent positions set.

the syntactic features do not need the parent pointers.. binding should add these. so

Reply:
LanguageService can be syntacticOnly right? I think thinking of it as disabled is wrong usage of term but even right now when we use that term we really mean that we will only allow syntactic features right?

>the syntactic features do not need the parent pointers.. binding should add these. so
If that's the assumption then we shouldn't need this code as well.. If we are creating ls for such things, we are better off passing that flag down to language service than tweaking this. no?